src/Compiler/Microsoft.AspNetCore.Razor.Language/src/BoundAttributeDescriptor.cs

```diff
@@ -91,7 +92,27 @@ public bool HasIndexer
         protected set => SetOrClearFlag(HasIndexerBit, value);
     }
 
-    public string Documentation { get; protected set; }
+    public string Documentation
+    {
+        get
+        {
+            return _documentationObject switch
+            {
+                string s => s,
+                DocumentationDescriptor d => d.GetText(),
+                null => null,
+                _ => throw new NotSupportedException()
+            };
```

Comment:
Perhaps we could have a helper for this chunk of code to avoid repetition?

Reply:
Yup! Thanks for the suggestion., I'm doing exactly that.